packages/dx-react-demos/src/demo-sources/grid-tree-data/hierarchical-tree.jsxt

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';<%&additionalImports%>
+import {
+  SelectionState,
+  TreeDataState,
+  CustomTreeData,
+  IntegratedSelection,
+} from '@devexpress/dx-react-grid';
+import {
+  Grid,
+  Table,
+  TableHeaderRow,
+  TableTreeColumn,
+} from '@devexpress/dx-react-grid-<%&themeName%>';
+
+import {
+  generateRows,
+  defaultColumnValues,
+} from '../../../demo-data/generator';
+
+const getChildRows = (row, rootRows) =>
+  (row ? row.items : rootRows);
+
+export default class Demo extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      columns: [
+        { name: 'name', title: 'Name' },
+        { name: 'sex', title: 'Sex' },
+        { name: 'city', title: 'City' },
+        { name: 'car', title: 'Car' },
+      ],
+      tableColumnExtensions: [
+        { columnName: 'name', width: 300 },
+      ],
+      data: generateRows({
```

Comment:
Why not extract a function (e.g. `generateTreeRows`) and place it in the generator? All this stuff related to tree data structure creation seems to be internal and does not have value to the demo itself.

Reply:
I don't like an idea to extract a function here. We inline such code in demos.